pkg/controllers/dnspolicy/dnspolicy_dnsrecords.go

```diff
@@ -45,42 +40,48 @@ func (r *DNSPolicyReconciler) reconcileDNSRecords(ctx context.Context, dnsPolicy
 	return nil
 }
 
-func (r *DNSPolicyReconciler) reconcileGatewayDNSRecords(ctx context.Context, gateway *gatewayv1beta1.Gateway, dnsPolicy *v1alpha1.DNSPolicy) error {
+func (r *DNSPolicyReconciler) reconcileGatewayDNSRecords(ctx context.Context, gw *gatewayv1beta1.Gateway, dnsPolicy *v1alpha1.DNSPolicy) error {
 	log := crlog.FromContext(ctx)
 
-	if err := r.dnsHelper.removeDNSForDeletedListeners(ctx, gateway); err != nil {
+	gatewayWrapper := utils.NewGatewayWrapper(gw)
+	err := gatewayWrapper.Validate()
+	if err != nil {
+		return err
+	}
```

Comment:
For consistency with the rest of this file, this check might be better written as:

```
gatewayWrapper := utils.NewGatewayWrapper(gw)
if err := gatewayWrapper.Validate(); err != nil {
	return err
}
```

Reply:
changed